### PR TITLE
[PHP] Replace `substr(PHP_OS)` calls with `PHP_OS_FAMILY` in PHP 7.2

### PR DESCRIPTION
**Pull request by Ayesh**
Trac ticket: https://core.trac.wordpress.org/ticket/61574
Follow-up to #6969, and responding to a comment in @jrfnl's #6975

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-ftp.php [hunk 1]
```diff
@@ -149,8 +149,11 @@ function __construct($port_mode=FALSE, $verb=FALSE, $le=FALSE) {
 	    $this->OS_local=FTP_OS_Unix;
 		$this->OS_remote=FTP_OS_Unix;
 		$this->features=array();
-		if(strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') $this->OS_local=FTP_OS_Windows;
-		elseif(strtoupper(substr(PHP_OS, 0, 3)) === 'MAC') $this->OS_local=FTP_OS_Mac;
+		if ( PHP_OS_FAMILY === 'Windows' ) {
+			$this->OS_local = FTP_OS_Windows;
+		} elseif ( PHP_OS_FAMILY === 'Darwin' ) {
+			$this->OS_local = FTP_OS_Mac;
+		}
```
Comment (jrfnl) [inline src/wp-admin/includes/class-ftp.php]: I'm not sure about this change. The `class-ftp.php` file is regarded as part of an [external dependency](https://www.phpclasses.org/package/1743-PHP-FTP-client-in-pure-PHP.html), so should generally not be touched unless to update the package.

Having said that, the external dependency is, AFAICS, no longer actively maintained... 🤷🏻‍♀️ 

Pro-tip: When in doubt about whether something is an external dependency or not, you can always check via the [`phpcs.xml.dist` file](https://github.com/WordPress/wordpress-develop/blob/f90c8bf1ae620b7568851f6d6e155223a8a6bf73/phpcs.xml.dist#L45-L78), as external dependencies don't have to comply with the WP Coding Standards.

Comment (Ayesh) [reply src/wp-admin/includes/class-ftp.php]: > Pro-tip: When in doubt about whether something is an external dependency or not, you can always check via the [phpcs.xml.dist file](https://github.com/WordPress/wordpress-develop/blob/f90c8bf1ae620b7568851f6d6e155223a8a6bf73/phpcs.xml.dist#L45-L78), as external dependencies don't have to comply with the WP Coding Standards.

TIL, thank you! 

I will remove this and push again.

Comment (Ayesh) [reply src/wp-admin/includes/class-ftp.php]: TIL, thank you!

I will remove this and push again.

**Review by jrfnl — APPROVED** [review]
LGTM! Thanks @Ayesh !

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r58684](https://core.trac.wordpress.org/changeset/58684).

**Pull request closed without merging**